apps/web-app/components/convos/upload.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+  /** Handles the file upload process for convo attachments.
+ * example usage:
+ * <ConvosUpload
+      v-model:uploadedAttachments="attachments"
+      :org-slug="orgSlug">
+      <template #default="{ openFileDialog, loading }">
+        <UnUiButton
+          :loading="loading"
+          label="refresh"
+          @click="openFileDialog" />
+      </template>
+    </ConvosUpload>
+ */
+  import { useFileDialog } from '@vueuse/core';
+
+  type Props = {
+    /**
+     * The organization slug required for the file upload process.
+     * @type {string}
+     */
+    orgSlug: string;
+  };
+  const props = defineProps<Props>();
+
+  /**
+   * Tracks the loading state of the file upload process.
+   * @type {Ref<boolean>}
+   */
+  const loading = ref(false);
+
+  /**
+   * provides the value of attachments uploaded by the user.
+   * @type {Ref<ConvoAttachmentUpload[]>}
+   */
+  const uploadedAttachments = defineModel<ConvoAttachmentUpload[]>(
+    'uploadedAttachments',
+    {
+      default: () => []
+    }
+  );
+
+  const {
+    open: openFileDialog,
+    onChange: selectedFilesOnChange,
+    reset: resetSelectedFiles
+  } = useFileDialog({
+    multiple: true
+  });
+
+  selectedFilesOnChange(async (selectedFiles) => {
+    if (!selectedFiles) return;
+    loading.value = true;
+
+    const storageUrl = useRuntimeConfig().public.storageUrl;
+
+    const uploadPromises = Array.from(selectedFiles).map(async (file: File) => {
+      const filename = file.name;
+      const fileType = file.type;
+
+      type PreSignedData = {
+        publicId: string;
+        signedUrl: string;
+      };
+      const { data: preSignedData } = await useFetch<PreSignedData>(
+        `${storageUrl}/api/attachments/presign`,
+        {
+          method: 'get',
+          params: {
+            orgSlug: props.orgSlug,
+            filename: filename
+          },
+          credentials: 'include'
+        }
+      );
```

Comment:
I would rather use [$fetch](https://nuxt.com/docs/api/utils/dollarfetch) here. We dont need a ref here since we dont do any reactivity here